Fusion Filtering Staggered PR7 (#42)
PediatricOpenTargets/OpenPBTA-analysis: analyses/fusion_filtering/README.md

All fusion calls have an additional column `reciprocal_exists` to specify if within the Sample a fusion GeneX--GeneY has a reciprocal GeneY--GeneX . `DomainRetainedGene1A` and `DomainRetainedGene1B` are added to identify kinase domain retention for Gene1A (5` Gene) and Gene1B (3` Gene).
Oncogene annotated fusions do not need to be in both callers to be retained however if these fusions are found in more than 4 broad histologies we treat them as false calls and remove them (this filtering step happen in the next step).
To scavenge back non-oncogenic fusions that are recurrently found uniquely in a cancer group we kept fusions that were called by both callers and if >2 samples per cancer group called the fusion.
We removed the non-oncogenic fusions with genes fused more than 5 times in a samples or found in more than 1 cancer group as potential artifact. 

> `05-QC_putative_onco_fusion_dustribution.Rmd` : Plots fusions found in multiple (more than 4) cancer groups in scratch/fusion-putative-oncogenic-preQC.tsv from 04-project-specific-filtering.Rmd. For final filtering, fusion calls found in more than 4 broad_histology groups are removed as QC filtering.

Reviewer: ```suggestion
`05-QC_putative_onco_fusion_distribution.Rmd` : Plots fusions found in multiple (more than 4) cancer groups in scratch/fusion-putative-oncogenic-preQC.tsv from 04-project-specific-filtering.Rmd. For final filtering, fusion calls found in more than 4 broad_histology groups are removed as QC filtering.
```

* fusion-recurrent-fusion-bysample.tsv
* fusion-recurrently-fused-genes-bycancergroup.tsv
* fusion-recurrently-fused-genes-bysample.tsv

### Run script
use OPENPBTA_BASE_SUBTYPING=1 to run this module using the histologies-base.tsv from data folder while running molecular-subtyping modules for release.
```sh
OPENPBTA_BASE_SUBTYPING=1 run_fusion_merged.sh 
```

OR by default uses histologies.tsv from data folder
```sh
bash run_fusion_merged.sh
```

#### Order of scripts in analysis
`00-normal-matrix-generation.R` : Based on the specific sample type, this script filter the gene expression file to contain only normal samples with the specific sample type

`01-fusion-standardization.R` : Standardizes fusion calls from STARFusion and Arriba

`02-fusion-filtering.R` : Filters artifacts by removing readthroughs and NEIGHBORS from annots column; annots column also contains red flag databases that are used to further filter common/normal occuring fusions; filters all fusions where both genes are expressed TPM < 1 are removed as non-expressed; requires fusions to have at least 1 JunctionSpanningRead and SpanningFragCount-JunctionReadCount to be <100

> `03-Calc-zscore-annotate.R` : Calculates z-score for gene fused gene's expression compared to GTeX normal samples. Please note that for different cohort, different normal expression files are used and hence the input for the cohort of interest and the input for the normal GTEx expression file needs to be in order

Reviewer 2: ```suggestion
`03-Calc-zscore-annotate.R` : Calculates z-score for gene fused gene's expression compared to GTeX normal samples. Please **note** that for different cohort, different normal expression files are used and hence the input values for the cohort of interest (--cohortInterest) and the input values for the normal GTEx expression file (--normalExpressionMatrix) need to be in the same order.
```

`04-project-specific-filtering.Rmd` : Performs project specific filtering. We prioritize the fusions as putative-oncogenic fusions if any fused gene in the fusion is annotated as kinases, oncogenes, tumor suppressors, curated transcription factors or present in COSMIC Cancer Gene Census list. We also annotated fusions if they are present in TCGA fusions list.
All fusion calls are additionally have columns `reciprocal_exists` to specify if within the Sample a fusion GeneX--GeneY has a reciprocal GeneY--GeneX . `DomainRetainedGene1A` and `DomainRetainedGene1B` are added to identify kinase domain retention for Gene1A (5` Gene) and Gene1B (3` Gene).
Oncogene annotated fusions do not need to be in both callers to be retained however if these fusions are found in more than 4 cancer groups we treat them as false calls and remove them.

> To scavenge back non-oncogenic fusions that are recurrently found uniquely in a cancer group we kept fusions that were called by both callers and if >2 samples per hcancer group called the fusion.

Reviewer 2: ```suggestion
To scavenge back non-oncogenic fusions that are recurrently found uniquely in a cancer group we kept fusions that were called by both callers and if >2 samples per cancer group called the fusion.
```
